search/utils.hpp

```diff
@@ -3,31 +3,125 @@
 #include "search/token_slice.hpp"
 
 #include "indexer/categories_holder.hpp"
+#include "indexer/search_delimiters.hpp"
+#include "indexer/search_string_utils.hpp"
 
 #include "base/buffer_vector.hpp"
+#include "base/levenshtein_dfa.hpp"
+#include "base/stl_helpers.hpp"
 #include "base/string_utils.hpp"
 
+#include <algorithm>
+#include <cctype>
+#include <functional>
+#include <queue>
+
 namespace search
 {
+namespace
```

Comment:
There's no sense in the anonymous namespace in the header for template function.